Ранее случайно сделал слияние веток на GitHaub

AlexanderKolnakov opened the pull request:
feat: - добавить новый функционал (обработка новых эндпоинтов) по добавлению/удалению друзей, выводу списка друзей, возможности ставить/снимать like фильмам, выводу списка наиболее популярных фильмов. refactor: -перенести функционал из контролеров в другие слои (Storage и Service) test: - переработать тесты в параметризованные
- тесты на новый функционал еще не добавлены

AlexanderKolnakov committed:
Ранее случайно сделал слияние веток на GitHaub

feat: - добавить новый функционал (обработка новых эндпоинтов) по добавлению/удалению друзей, выводу списка друзей, возможности ставить/снимать like фильмам, выводу списка наиболее популярных фильмов.
refactor: -перенести функционал из контролеров в другие слои (Storage и Service)
test: - переработать тесты в параметризованные
- тесты на новый функционал еще не добавлены

EvgenyKoptubenko reviewed (changes requested):
Есть несколько замечаний, но в PR нет всех файлов.

EvgenyKoptubenko commented:
UserService в функции showGeneralFriends можно использовать функцию showFriend - дублирование функциональности

EvgenyKoptubenko commented:
Замечаний по коду вообще нет. Отлично все сделано.
Есть одно предложение из разряда "Можно лучше":
Попробуй сделать централизованную обработку исключений https://www.baeldung.com/exception-handling-for-rest-with-spring (Есть еще @RestControllerAdvice)

AlexanderKolnakov committed:
feat: - разработать и добавить в README схему БД

AlexanderKolnakov committed:
refactor: - удалить дублирующийся код в методе showGeneralFriends класса UserService
- переработать исключение, сделать центральную обработку исключение с использованием @RestControllerAdvice

feat: - добавить схему БД в README.md (в рамках ТЗ 12 спринта это БД была проверена сокурсником)

EvgenyKoptubenko reviewed (approved):
Работа сделана очень хорошо.
